apps/backoffice/src/lib/be/cms-client.ts

```diff
@@ -0,0 +1,39 @@
+import { createClient } from "@/generated/services-cms/client";
+import { BooleanFromString } from "@pagopa/ts-commons/lib/booleans";
+import { readableReport } from "@pagopa/ts-commons/lib/reporters";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import * as E from "fp-ts/lib/Either";
+import * as t from "io-ts";
+import { cache } from "react";
+
+const Config = t.type({
+  API_SERVICES_CMS_URL: NonEmptyString,
+  API_SERVICES_CMS_BASE_PATH: NonEmptyString,
+  API_SERVICES_CMS_MOCKING: BooleanFromString
+});
+
+const getIoServicesCmsClientConfig = cache(() => {
+  const result = Config.decode(process.env);
+
+  if (E.isLeft(result)) {
+    throw new Error("error parsing io-services-cms client config", {
+      cause: readableReport(result.left)
+    });
+  }
+  return result.right;
+});
+
+export const getIoServicesCmsClient = cache(() => {
+  const configuration = getIoServicesCmsClientConfig();
+
+  if (configuration.API_SERVICES_CMS_MOCKING) {
+    const { setupMocks } = require("../../../mocks");
+    setupMocks();
+  }
+
+  return createClient({
+    baseUrl: configuration.API_SERVICES_CMS_URL,
+    fetchApi: (fetch as any) as typeof fetch,
+    basePath: configuration.API_SERVICES_CMS_BASE_PATH
+  });
+});
```

Comment:
suggestion (non-blocking): remove cache from _getIoServicesCmsClientConfig_.

There is no need to cache _getIoServicesCmsClientConfig_, because it is not exported and it will be called only once by _getIoServicesCmsClient_. So, in this way the only fn to be cached will be _getIoServicesCmsClient_.

Reply:
4f472fbf93f8f626f54e4a2a415e2a156c1cc195